src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,69 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import { UrlSearchParams } from '../types/UrlSearchParams';
+
+const centuriesOnPage = ['16', '17', '18', '19', '20'];
+
+const emptyParams = {
+  sex: null,
+  query: null,
+  centuries: null,
+};
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get(UrlSearchParams.Query) || '';
+  const centuriesFromUrl = searchParams.getAll(UrlSearchParams.Centuries) || [];
+
+  function handleQueryChange(e: React.ChangeEvent<HTMLInputElement>) {
+    const params = new URLSearchParams(searchParams);
+
+    if (e.target.value.trim().length === 0) {
+      params.delete(UrlSearchParams.Query);
+    } else {
+      params.set(UrlSearchParams.Query, e.target.value);
+    }
+
+    setSearchParams(params);
+  }
+
+  function handleCenturiesChange(century: string) {
+    return centuriesFromUrl.includes(century)
+      ? centuriesFromUrl.filter(currentCentury => currentCentury !== century)
+      : [...centuriesFromUrl, century];
+  }
```

Comment:
The handleCenturiesChange function returns a new array of centuries but does not update the search parameters directly. Ensure that this function is used to update the search parameters correctly, possibly by integrating it with setSearchParams.

Reply:
It works the same way with the help of SearchLink.tsx component